azureml/http.py

```diff
@@ -1,198 +1,213 @@
+    SESSION_ID_HEADER_NAME = 'x-ms-client-session-id'
+    SESSION_ID_HEADER_VALUE = 'DefaultSession'
+    ACCESS_TOKEN_HEADER_NAME = 'x-ms-metaanalytics-authorizationtoken'
+    CONTENT_TYPE_HEADER_NAME = 'Content-Type'
+    CONTENT_TYPE_HEADER_VALUE_JSON = 'application/json;charset=UTF8'
+    CHUNK_SIZE = 0x200000 
+    DEFAULT_OWNER = 'Python SDK'
+    USER_AGENT_HEADER_NAME = 'User-Agent'
+    USER_AGENT_HEADER_VALUE = 'pyazureml/' + __version__
+
+    def __init__(self, service_endpoint, access_token):
+        self._service_endpoint = service_endpoint
+        self._access_token = access_token
+
+    def get_experiments(self, workspace_id):
+        """Runs HTTP GET request to retrieve the list of experiments."""
+        api_path = self.EXPERIMENTS_URI_FMT.format(workspace_id)
+        return self._send_get_req(api_path)
+
+    def get_datasets(self, workspace_id):
+        """Runs HTTP GET request to retrieve the list of datasets."""
+        api_path = self.DATASOURCES_URI_FMT.format(workspace_id)
+        return self._send_get_req(api_path)
+
+    def get_dataset(self, workspace_id, dataset_id):
+        """Runs HTTP GET request to retrieve a single dataset."""
+        api_path = self.DATASOURCE_URI_FMT.format(workspace_id, dataset_id)
+        return self._send_get_req(api_path)
+
+    def open_intermediate_dataset_contents(self, workspace_id, experiment_id,
+                                           node_id, port_name):
+        return self._get_intermediate_dataset_contents(
+            workspace_id,
+            experiment_id,
+            node_id,
+            port_name,
+            stream=True).raw
+
+    def read_intermediate_dataset_contents_binary(self, workspace_id,
+                                                  experiment_id, node_id,
+                                                  port_name):
+        return self._get_intermediate_dataset_contents(
+            workspace_id,
+            experiment_id,
+            node_id,
+            port_name,
+            stream=False).content
+
+    def read_intermediate_dataset_contents_text(self, workspace_id,
+                                                experiment_id, node_id,
+                                                port_name):
+        return self._get_intermediate_dataset_contents(
+            workspace_id,
+            experiment_id,
+            node_id,
+            port_name,
+            stream=False).text
+
+    def _get_intermediate_dataset_contents(self, workspace_id, experiment_id,
+                                           node_id, port_name, stream):
+        api_path = self.INTERMEDIATE_DATASET_URI_FMT.format(
+            workspace_id, experiment_id, node_id, port_name)
+        response = requests.get(
+            url=urljoin(self._service_endpoint, api_path),
+            headers=self._get_headers(),
+            stream=stream,
+        )
+        return response
+
+    def open_dataset_contents(self, url):

```

Comment:
On all three of these overloads we should raise an exception if the response > 400